src/vos/vos_obj_cache.c

```diff
@@ -408,6 +408,13 @@ vos_obj_hold(struct daos_lru_cache *occ, struct vos_container *cont,
 		goto failed;
 	}
 
+	if (obj->obj_discard && intent == DAOS_INTENT_UPDATE) {
+		/** Cleanup before assert so unit test that triggers doesn't corrupt the state */
+		vos_obj_release(occ, obj, false);
+		rc = -DER_UPDATE_AGAIN;
```

Comment:
here for UPDATE will return update again, but above L404, for "create || (flags & VOS_OBJ_DISCARD) != 0)" it directly fail it, but for L404's case it does not set err code, not very sure if any problem? @wangdi1 @jolivier23

Reply:
I don't see how we would ever get here.   If intent is DAOS_INTENT_UPDATE, wouldn't we assert on line 407 because "create" should be true?